src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,512 @@
+
+    leFilter->setValidator(new PrimerValidator(this, false));
+    connect(leFilter, &QLineEdit::textEdited, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateSequenceList);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_activeSequenceChanged() {
+    makeWarningInvisibleIfDna();
+    sl_sequenceModified();
+
+    // Release "Select manually" buttons.
+    tbLeftAreaSelectManually->setChecked(false);
+    tbRightAreaSelectManually->setChecked(false);
+    smButton = nullptr;
+    sbStartRegion = nullptr;
+    sbEndRegion = nullptr;
+    disconnect(updateRegionConnection);
+    updateRegionConnection = QMetaObject::Connection();
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_sequenceModified() {
+    ADVSequenceObjectContext *sequenceContext = annDnaView->getActiveSequenceContext();
+    CHECK(sequenceContext != nullptr, )
+    qint64 seqLength = sequenceContext->getSequenceLength();
+
+    // End spinbox maximum value is the previous sequence length. If it has changed, need to update spinboxes.
+    if (seqLength != sbLeftAreaEnd->maximum()) {
+        int start = sbLeftAreaStart->value() - 1;
+        int len = sbLeftAreaEnd->value() - start;
+        setRegion(sbLeftAreaStart, {start, len});
+
+        start = sbRightAreaStart->value() - 1;
+        len = sbRightAreaEnd->value() - start;
+        setRegion(sbRightAreaStart, {start, len});
+    }
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_start() {
+    PCRPrimerDesignForDNAAssemblyTaskSettings settings;
+    settings.forwardUserPrimer = leForwardPrimer->text();
+    settings.reverseUserPrimer = leReversePrimer->text();
+
+    settings.gibbsFreeEnergy.minValue = sbMinRequireGibbs->value();
+    settings.gibbsFreeEnergy.maxValue = sbMaxRequireGibbs->value();
+    settings.meltingPoint.minValue = spMinRequireMeltingTeml->value();
+    settings.meltingPoint.maxValue = spMaxRequireMeltingTeml->value();
+    settings.overlapLength.minValue = spMinRequireOverlapLength->value();
+    settings.overlapLength.maxValue = spMaxRequireOverlapLength->value();
+
+    settings.gibbsFreeEnergyExclude = sbExcludeGibbs->value();
+    settings.meltingPointExclude = spExcludeMeltingTeml->value();
+    settings.complementLengthExclude = spExcludeComplementLength->value();
+
+    if (backbone5->isChecked()) {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5;
+    } else {
+        settings.insertTo = PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3;
+    }
+    settings.bachbone5Length = sbBackbone5Length->value();
+    settings.bachbone3Length = sbBackbone3Length->value();
+
+    settings.leftArea.startPos = sbLeftAreaStart->value() - 1;
+    settings.leftArea.length = sbLeftAreaEnd->value() - sbLeftAreaStart->value();
+    settings.rightArea.startPos = sbRightAreaStart->value() - 1;
+    settings.rightArea.length = sbRightAreaEnd->value() - sbRightAreaStart->value();
```

Comment:
For length, replace `sbLeftAreaStart->value()` with `settings.leftArea.startPos`. The same for `rightArea`. If you did, replace `area.length >= 0` with `area.length > 0` at https://github.com/ugeneunipro/ugene/blob/59e80071b0587d3a2022382a45b1d03f05c406b9/src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp#L240